packages/payment/src/paymentGateway.ts

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) 2023 System Automation Corporation.
+// This file is licensed under the MIT License.
+
+import { Payment, PaymentResult } from './payment';
+
+export interface ParsedQueryString {
+    [key: string]: undefined | string | string[] | ParsedQueryString | ParsedQueryString[];
+}
+export interface TransferData {
+    method?: 'GET' | 'POST';
+    url: string;
+    parameters: Record<string, string>;
+}
+export interface PaymentGateway {
+    prepare(payment: Payment, returnUrl: string): TransferData | PromiseLike<TransferData>;
+    postPaymentResult(
+        payment: Payment,
+        resultData: ParsedQueryString,
+    ): PaymentResult | null | PromiseLike<PaymentResult | null>;
```

Comment:
In ML1, PaymentResult was a different type, whereas in Evoke it's all in the same Payment object.  What do you think about having `postPaymentResult` just updating `payment` (or returns the updated payment)?  It seems weird for `Payment` to extend `PaymentResult` (basically saying every payment is a result, which doesn't really make sense).  Also `status` is defined on `PaymentResult`, but really only 'Declined', 'Cancelled', and 'Paid' are really "result" statuses.

Reply:
Hmm. I like the idea of having `postPaymentResult` return an updated `payment`, and we can then update the instance that we store to match.